src/aliceVision/keyframe/README.md

```diff
@@ -0,0 +1,135 @@
+
+## Smart selection method
+
+The smart selection works in two steps:
+- for each frame in the input video / sequence, a sharpness score and a motion score are computed;
+- the sharpness and motion scores are used as well as the temporal position of the evaluated frame to determine whether the frame will be selected.
+
+The method aims at selecting a frame that is as sharp as possible with significant motion compared to the previously selected frame: consecutive frames should not be selected as keyframes if they do not contain enough motion, even if they are both very sharp.
+
+The minimum and maximum number of selected keyframes with the smart method can be set with the following parameters:
+- `minNbOutFrames`: the minimum number of selected keyframes;
+- `maxNbOutFrames`: the maximum number of selected keyframes.
+
+### Frame scoring
+
+For both the sharpness and motion scores, the evaluated frame is converted to a grayscale OpenCV matrix that may be rescaled
+Scores are computed on grayscale images, which may have been rescaled using the `rescaledWidth` parameter.
+
+#### Sharpness score
+
+The Laplacian of the input frame is first computed, followed by the integral image of the Laplacian. A sliding window of size `sharpnessWindowSize` is used to compute the standard deviation of the averaged Laplacian locally. The final sharpness score will be the highest standard deviation found.
+
+The image is evaluated with a sliding window instead of as a whole to prevent giving a bad score (low standard deviation) to a frame that contains a sharp element but is overall blurry.
+
+#### Motion score
+
+The dense optical flow of a frame is computed. The frame is then divided into cells of `flowCellSize` pixels in which the motion vectors are averaged to obtain a displacement value (in pixels) within that cell. Once all the displacement values have been computed, the median value of these displacement values is used as the motion score.
+
+### Selection
+
+Once both the sharpness and motion scores have been computed, subsequences are identified based on the motion accumulation across frames. The motion accumulation threshold is set with `pxDisplacement` which represents, in percent, the amount of pixels that need to have moved since the last keyframe for the motion to be significant. As the motion scores represent a displacement value for each frame, summing them over time until the accumulation reaches the threshold allows to divide the input video / sequence into subsequences that all contain significant motion.
```

Comment:
```suggestion
Once both the sharpness and motion scores have been computed, subsequences are identified based on the motion accumulation across frames. The motion accumulation threshold is set with `pxDisplacement` which represents, in per cent, the number of pixels that need to have moved since the last keyframe for the motion to be significant. As the motion scores represent a displacement value for each frame, summing them over time until the accumulation reaches the threshold allows to divide the input video / sequence into subsequences that all contain significant motion.
```

Reply:
Done, thanks!